src/microbe_stage/MembraneWaterRipple.cs

```diff
@@ -0,0 +1,666 @@
+    private readonly StringName phaseParam = new("Phase");
+    private readonly StringName attenuationParam = new("Attenuation");
+    private readonly StringName pastPositionsParam = new("PastPositions");
+    private readonly StringName pastPositionsCountParam = new("PastPositionsCount");
+    private readonly StringName stillnessFactorParam = new("StillnessFactor");
+    private readonly StringName membraneRadiusParam = new("MembraneRadius");
+
+    /// <summary>
+    ///   Fade-in speed multiplier (higher = faster fade-in)
+    /// </summary>
+    [Export]
+    private float fadeInSpeed = 7.0f;
+
+    /// <summary>
+    ///   Fade-out speed multiplier
+    /// </summary>
+    [Export]
+    private float fadeOutSpeed = 0.6f;
+
+    [Export]
+    private MeshInstance3D waterPlane = null!;
+
+    private ShaderMaterial waterMaterial = null!;
+    private PlaneMesh planeMesh = null!;
+
+    /// <summary>
+    ///   Default Size of the membrane we're creating ripples for
+    /// </summary>
+    private float membraneRadius = 5.0f;
+
+    /// <summary>
+    ///   Parent node for positioning
+    /// </summary>
+    private Node3D? positionTarget;
+
+    /// <summary>
+    ///   Position tracking and effect state variables
+    /// </summary>
+    private Vector2[] pastPositions = new Vector2[14];
+
+    private Vector3[] positionHistory = new Vector3[14];
+    private Array<Vector2> godotPastPositions = new();
+    private int currentPositionIndex;
+    private bool isPositionHistoryFull;
+    private float positionRecordTimer;
+    private Vector3 lastPosition;
+    private Vector3 previousPosition;
+    private Vector2 currentDirection = Vector2.Right;
+    private Vector2 lastValidDirection = Vector2.Right;
+    private float currentSpeed;
+    private float timeAccumulator;
+    private bool isCurrentlyVisible = true;
+    private float visibilityCheckTimer;
+    private bool isEffectEnabled = true;
+
+    /// <summary>
+    ///   Stillness tracking variables
+    /// </summary>
+    private float stillnessTimer;
+
+    private float stillnessFactor;
+    private bool wasMovingLastFrame;
+    private float averageMovementSqr;
+    private float directionChangeTimer;
+    private float lastDirectionChangeTime;
+    private float timeWithoutMovement;
+
+    /// <summary>
+    ///   Ripple effect variables
+    /// </summary>
+    private float targetAlpha;
+
+    private float currentAlpha;
+    private float stillnessValue = 1.0f;
+    private float targetStillness = 1.0f;
+    private bool isForming;
+    private float formingTime;
+    private float minAlpha = 0.00002f;
+    private float fullAlpha = 0.02f;
+
+    /// <summary>
+    ///   Camera state caching variables
+    /// </summary>
+    private Camera3D? currentCamera;
+
+    private float lastCameraDistance;
+    private Vector3 lastCameraPosition;
+    private bool isCameraPositionValid;
+
+    // initialization flag
+    private bool isInitialized;
+#pragma warning restore CA2213
```

Comment:
There's a bunch of normal primitive type variables that should absolutely not be inside the warning disable block